stream/stream.rkt

```diff
@@ -0,0 +1,24 @@
+#lang racket
+
+(require "cons.rkt")
+
+;kstream instead of stream because racket already has a stream
+;a stream is basically lazy evaluation done manually
+;when evaluated the tail of stream should produce another stream
+;which is the next value + a promisse to calculate the rest
+
+(define (kstream head tail)
+  (kons head tail))
+  
+(define (kstream-head s)
+  (kons-head s))
+  
+(define (kstream-tail s)
+  ((kons-tail s)))
```

Comment:
I would not use this syntax because it's very misleading, don't you think ? I stared at the code for some time finding the promise evaluator. Maybe my eyes are not tuned for lisp anymore =P But I would use `apply` to improve readability anyway.

Reply:
If I recall correctly, Alberd in the Stream lectures use that syntax for double funcalls in the delay/force system.

Reply:
In the course the ((func)) is used, but I agree with @tiago4orion that it may hurt the eyes indeed. Will change to apply just to make it more explicit =)